enumerables.rb

```diff
@@ -0,0 +1,171 @@
+# rubocop:disable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity,Metrics/ModuleLength
+#  rubocop:disable Metrics/MethodLength,Style/CaseEquality
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    each do |item|
+      yield item
+    end
+
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    index = 0
+    each do |item|
+      yield item, index
+      index += 1
+    end
+
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    results_array = []
+    each do |item|
+      next unless yield item
+
+      results_array << item
+    end
+    results_array
+  end
+
+  def my_all?(arg = nil)
+    result = true
+    if block_given?
+      each do |item|
+        result = false unless yield item
+      end
+    elsif arg
+      if arg.is_a?(Regexp)
+        each do |item|
+          result = false unless arg.match(item)
+        end
+      else
+        each do |item|
+          result = false unless arg === item
+        end
+      end
+    else
+      each do |item|
+        result = false unless item
+      end
+    end
+    result
+  end
+
+  def my_any?(arg = nil)
+    result = false
+    if block_given?
+      each do |i|
+        result = true if yield i
+
+        next
+      end
+    elsif arg
+      if arg.is_a?(Regexp)
+        each do |item|
+          result = true if arg.match(item)
+        end
+      else
+        each do |item|
+          result = true if arg === item
+        end
+      end
+    else
+      each do |i|
+        result = true if i
+
+        next
+      end
+    end
+    result
+  end
+
+  def my_none?(arg = nil)
+    result = true
+    if block_given?
+      each do |item|
+        result = false if yield item
+      end
+    elsif arg
+      if arg.is_a?(Regexp)
+        each do |item|
+          result = false if arg.match(item)
+        end
+      else
+        each do |item|
+          result = false if arg == item
+        end
+      end
+    else
+      each do |item|
+        result = false if item
+      end
+    end
+    result
+  end
+
+  def my_count(number = nil)
+    input = Array(self)
+
+    count = 0
+    if number
+      each do |i|
+        count += number if i == number
+      end
+    end
+
+    return input.length unless block_given?
+
+    each do |i|
+      count += 1 if yield i
+    end
+    count
+  end
```

Comment:
# my_count
- Please ensure that it counts the number of items in enum that are equal to the item if an argument is given.
In my testing variables, 
`Hint` I expected =>  `expect(array.my_count(LOWEST_VALUE)).to eq array.count(LOWEST_VALUE)` to give me 5 but I got 100.